src/controllers/referral.js

```diff
@@ -25,24 +20,23 @@ const {
   COMPLETED_SUCCESSFULLY,
 } = messages;
 
-const getSubjectUpi = async (dataRecord) => {
-  logger.information(DATA_RECORD);
-  logger.information(JSON.stringify(dataRecord));
-  let upi = dataRecord.upi;
-
-  if (!upi) {
-    const echisDoc = await getEchisDocForUpdate(dataRecord._patient_id);
-    upi = echisDoc.upi;
+const getSubjectUpi = async (echisClientId) => {
+  try {
+    const echisClient = await getDoc({ instance: CHT.url, user: CHT.username, password: CHT.password }, echisClientId);
+    if (echisClient.upi) {
+      return echisClient.upi;
+    }
```

Comment:
What should happen when there is no `upi`, but no error to be caught?

Reply:
We try and create the client in the next code block and if that fails then the function returns nothing